src/DataTypes/getLeastSupertype.h

```diff
@@ -18,7 +18,7 @@ enum class LeastSupertypeOnError
   * Examples: there is no least common supertype for Array(UInt8), Int8.
   */
 template <LeastSupertypeOnError on_error = LeastSupertypeOnError::Throw>
-DataTypePtr getLeastSupertype(const DataTypes & types);
+DataTypePtr getLeastSupertype(const DataTypes & types, bool optimize_type_ids = false);
```

Comment:
I think we don't need the new flag here, let's just do it always, it should not brake anything